source-firestore/discovery.go

```diff
@@ -68,6 +68,7 @@ type documentMetadata struct {
 	CreateTime *time.Time `json:"ctime,omitempty" jsonschema:"title=Create Time,description=The time at which the document was created. Unset if the document is deleted."`
 	UpdateTime *time.Time `json:"mtime" jsonschema:"title=Update Time,description=The time at which the document was most recently updated (or deleted)."`
 	Deleted    bool       `json:"delete,omitempty" jsonschema:"title=Delete Flag,description=True if the document has been deleted, unset otherwise."`
+	Backfill   bool       `json:"backfill,omitempty" jsonschema:"title=Backfill Flag,description=True if this document was captured via a backfill."`
```

Comment:
We use the term `snapshot` elsewhere: https://github.com/estuary/connectors/blob/e516b35943186765f73945b07651b1a5569f32f2/sqlcapture/interface.go#L35

Might be good to use the same terminology here for consistency?

Reply:
Yeah, you're probably right. It's a little ambiguous because the only reason we call it `snapshot` in other connectors is because that's how Debezium names that bit of metadata, and in most of our system and docs we're pretty consistent about naming that concept 'backfill' instead. But we're still using the name in newer connectors such as `source-snowflake` so I guess we ought to here as well.

Reply:
Done